### PR TITLE
Rethink Health::Patient.needs_qa

### PR DESCRIPTION
**Pull request by awisema**
## Description

Bugfixes for problems introduced when updating My Patient filters.

## Type of change
- [X] Bug fix
- [ ] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [X] My code follows the style guidelines of this project (rubocop)
- [X] I have updated the documentation (or not applicable)
- [X] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: app/controllers/concerns/health_patient_dashboard.rb [hunk 1]
```diff
@@ -49,14 +49,12 @@ def determine_sort_order(medicaid_ids, column, direction)
     end
 
     # @param patients Initial patients scope
-    # @param search_string|nil Search string
     # @param Hash<String>|nil Filters
     # @return patient_scope, search_scope, active_filter
-    def apply_filter(patients, search_string, filter)
-      search = search_setup(scope: :full_text_search)
-      patients = search.distinct if search_string.present?
+    def apply_filter(patients, filter)
```
Comment (eanders) [inline app/controllers/concerns/health_patient_dashboard.rb:54]: Are there other calls to 'apply_filter' that might need updating?

**Review by eanders — APPROVED** [review]
This looks good to me, assuming you got all instances of `apply_filter` and that the change to the logic makes sense.  Looks like some inverted `not`s.

**Comment by eanders** [comment]
@ttoomey I did it again(I think).  There's got to be a better way, somehow to prepare the PR, or set some rules about what should be squash-merged vs merged.  Thankfully I haven't yet squash merged a release (or haven't in a long time.)

**Comment by ttoomey** [comment]
@eanders no worries. It's hard because GH interface remembers whatever you did last. Wish they made that a prompt or something instead.

I suppose one approach would be to disable everything except squash merges for PRs (under Settings -> General -> Pull Requests). Then you'd have to merge releases via the CLI (`git merge release-91`). That might be too awkward though

**Comment by eanders** [comment]
There's also an option to require linear history on a protected branch.  Not sure if that would help or not though.

**Comment by ttoomey** [comment]
> There's also an option to require linear history on a protected branch. Not sure if that would help or not though.

Interesting but yeah, I don't know that it would solve this issue. We'd need a way to set merge strategy per branch rather than per repository.

**Pull request merged**